### PR TITLE
fix(Modal): 修复 modal 蒙层关闭逻辑

### PR DESCRIPTION
**Pull request by zym19960704**
**What kind of change does this PR introduce?** (check at least one)

-   [x] Bugfix
-   [ ] Feature
-   [ ] Code style update
-   [ ] Refactor
-   [ ] Docs
-   [ ] Build-related changes
-   [ ] Other, please describe:

**Does this PR introduce a breaking change?** (check one)

-   [ ] Yes
-   [x] No

If yes, please describe the impact and migration path for existing applications:

**Related issue (if exists):**

**Other information:**

### CODE REVIEW

File: components/modal/modal.tsx [hunk 1]
```diff
@@ -218,18 +236,21 @@ const Modal = defineComponent({
...
-                                }
+                                onClick={(event) => maskClick(event)}
                             >
                                 <div
                                     class={`${prefixCls}-wrapper ${
                                         props.contentClass || ''
                                     }`}
                                     style={styles.value}
                                     onClick={(event) => event.stopPropagation()}
+                                    onMousedown={() => {
```
Comment (ocean-gao) [inline components/modal/modal.tsx:247]: 看下这里停止冒泡可以吗？

File: components/modal/modal.tsx [hunk 2]
```diff
@@ -218,18 +236,21 @@ const Modal = defineComponent({
                                     [`${prefixCls}-no-footer`]: !props.footer,
                                 }}
                                 style={{ zIndex: zIndex.value }}
-                                onClick={(event) =>
-                                    props.maskClosable &&
-                                    props.mask &&
-                                    handleCancel(event)
-                                }
+                                onClick={(event) => maskClick(event)}
```
Comment (ocean-gao) [inline components/modal/modal.tsx:239]: handleClickMask

File: components/modal/modal.tsx [hunk 3]
```diff
@@ -218,18 +236,21 @@ const Modal = defineComponent({
...
                                         props.contentClass || ''
                                     }`}
                                     style={styles.value}
                                     onClick={(event) => event.stopPropagation()}
+                                    onMousedown={() => {
+                                        mouseDownInsideChild.value = true;
+                                    }}
+                                    onMouseup={() => {
+                                        mouseUpInsideChild.value = true;
+                                        mouseDownInsideChild.value = false;
```
Comment (ocean-gao) [inline components/modal/modal.tsx:252]: mouseDownInsideChild.value = false;
mouseUpInsideChild.value = true;
这两行换下顺序

File: components/modal/modal.tsx [hunk 4]
```diff
@@ -183,6 +183,24 @@ const Modal = defineComponent({
                 props.displayDirective === 'show',
         );
 
+        // 鼠标在弹窗内按下
+        const mouseDownInsideChild = ref(false);
+        // 鼠标在弹窗内弹起
+        const mouseUpInsideChild = ref(false);
```
Comment (1zumii) [inline components/modal/modal.tsx:189]: 好像只有赋值的操作，并没有把这个值拿来使用

Comment (zym19960704) [reply components/modal/modal.tsx:189]: mouseUpInsideChild 这个确实没用，已去掉

**Pull request merged**